Added javadoc and readme details for batch queries (#8)
target/lite-for-jdbc: README.md

## executeBatch

```kotlin
executeBatch(
    sql: String, 
    args: List<Map<String, Any?>>,
    rowMapper: RowMapper<T>
): List<T>
```

executeBatch is used to run the same SQL statement with different parameters in batch mode. 
This can give you significant performance improvements.

Args is a list of maps. Each item in the list will be a query execution in a batch. The Map will provide the parameters 

> for that execution. In the following example there will be two queries executed in a single bath. The first will 

Reviewer: ```suggestion
for that execution. In the following example there will be two queries executed in a single batch. The first will 
```

insert model1, and the second will insert model2.

> RowMapper will map the results to an object. Use this if the query is generating keys or otherwise selecting results.

Reviewer: ```suggestion
RowMapper maps the results to the specified result type. 
```

> The response is a list of Objects T. Each object will represent a result returned from a batch query. Most likely there will

Reviewer: ```suggestion
The response is a list of Objects of type `T`. Each object represents a batch query result. Most likely there will
```

> be one result per query execution. In the following example the results list will have 2 elements. The first element will 

Reviewer: ```suggestion
be one result per query execution. In the following example the results list will have 2 elements. The first element  
```

> provide the generated ID of the model1 object, the second element will provide the generated ID of the model2 object. 

Reviewer: ```suggestion
provides the generated ID of the model1 object, and the second element provides the generated ID of the model2 object. 
```

```kotlin

> val model1 = Model(field1 = "testName1", field2 = 1001)
> val model2 = Model(field1 = "testName2", field2 = 1002)
> val results = db.executeBatch(sql = "INSERT INTO T (field1, field2) VALUES (:field1, :field2)",
>   args = listOf(model1.propertiesToMap(), model2.propertiesToMap()),
>   rowMapper = { resultSet -> resultSet.get("id") }
> )
> 
> val newModel1 = model1.copy(id = results[0])
> val newModel2 = model2.copy(id = results[1])

Reviewer: ```suggestion
val models = listOf(
    Model(field1 = "testName1", field2 = 1001),
    Model(field1 = "testName2", field2 = 1002)
)

val insertedIds = db.executeBatch(
    sql = "INSERT INTO T (field1, field2) VALUES (:field1, :field2)",
    args = models.map { it.propertiesToMap() },
    rowMapper = { resultSet -> resultSet.get("id") }
)
```

```

## executeBatch Counts only

```kotlin
executeBatch(
    sql: String, 
    args: List<Map<String, Any?>>
): List<Int>
```

executeBatch is used to run the same SQL statement with different parameters in batch mode.
This can give you significant performance improvements.

Args is a list of maps. Each item in the list will be a query execution in a batch. The Map will provide the parameters

> for that execution. In the following example there will be two queries executed in a single bath. The first will

Reviewer: similar language suggestions as the above section, mostly making the sentences be more active
